sweepbatcher/sweep_batcher_test.go

```diff
@@ -245,6 +251,123 @@ func testSweepBatcherBatchCreation(t *testing.T, store testStore,
 	require.True(t, batcherStore.AssertSweepStored(sweepReq3.SwapHash))
 }
 
+// testFeeBumping tests that sweep is RBFed with slightly higher fee rate
+// after each block unless WithNoBumping is passed.
+func testFeeBumping(t *testing.T, store testStore,
+	batcherStore testBatcherStore, noFeeBumping bool) {
+
+	defer test.Guard(t)()
+
+	lnd := test.NewMockLnd()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sweepStore, err := NewSweepFetcherFromSwapStore(store, lnd.ChainParams)
+	require.NoError(t, err)
+
+	// Disable fee bumping, if requested.
+	var opts []BatcherOption
+	if noFeeBumping {
+		opts = append(opts, WithNoBumping())
+	}
+
+	batcher := NewBatcher(lnd.WalletKit, lnd.ChainNotifier, lnd.Signer,
+		testMuSig2SignSweep, testVerifySchnorrSig, lnd.ChainParams,
+		batcherStore, sweepStore, opts...)
+	go func() {
+		err := batcher.Run(ctx)
+		checkBatcherError(t, err)
+	}()
+
+	spendChan := make(chan *SpendDetail, ntfnBufferSize)
+
+	// Generate keys.
+	_, senderPubKey := test.CreateKey(1)
+	var senderKey [33]byte
+	copy(senderKey[:], senderPubKey.SerializeCompressed())
+	_, receiverPubKey := test.CreateKey(2)
+	var receiverKey [33]byte
+	copy(receiverKey[:], receiverPubKey.SerializeCompressed())
+
+	// Create a sweep request.
+	sweepReq1 := SweepRequest{
+		SwapHash: lntypes.Hash{1, 1, 1},
+		Value:    1_000_000,
+		Outpoint: wire.OutPoint{
+			Hash:  chainhash.Hash{1, 1},
+			Index: 1,
+		},
+		Notifier: &SpendNotifier{
+			SpendChan:    spendChan,
+			SpendErrChan: make(chan error, ntfnBufferSize),
+			QuitChan:     make(chan bool, ntfnBufferSize),
+		},
+	}
+
+	swap1 := &loopdb.LoopOutContract{
+		SwapContract: loopdb.SwapContract{
+			CltvExpiry:      111,
+			AmountRequested: 1_000_000,
+			ProtocolVersion: loopdb.ProtocolVersionMuSig2,
+			HtlcKeys: loopdb.HtlcKeys{
+				SenderScriptKey:        senderKey,
+				ReceiverScriptKey:      receiverKey,
+				SenderInternalPubKey:   senderKey,
+				ReceiverInternalPubKey: receiverKey,
+				ClientScriptKeyLocator: keychain.KeyLocator{
+					Family: 1,
+					Index:  2,
+				},
+			},
+		},
+
+		DestAddr:        destAddr,
+		SwapInvoice:     swapInvoice,
+		SweepConfTarget: 111,
+	}
+
+	err = store.CreateLoopOut(ctx, sweepReq1.SwapHash, swap1)
+	require.NoError(t, err)
+	store.AssertLoopOutStored()
+
+	// Deliver sweep request to batcher.
+	require.NoError(t, batcher.AddSweep(&sweepReq1))
+
+	// Since a batch was created we check that it registered for its primary
+	// sweep's spend.
+	<-lnd.RegisterSpendChannel
+
+	// Wait for sweep tx to appear in the wallet.
+	require.Eventually(t, func() bool {
```

Comment:
nit: there's no need for the eventually check here and below, since the tx can be read from `lnd.TxPublishChannel` and the guard will fail if the test timeouts.

Reply:
Thanks! Fixed.